runrestic/restic/tools.py

```diff
@@ -75,33 +103,45 @@ def log_messages(process: Any, proc_cmd: str) -> str:
 
 
 def retry_process(
-    cmd: Union[str, List[str]],
-    config: Dict[str, Any],
-    abort_reasons: Optional[List[str]] = None,
-) -> Dict[str, Any]:
+    cmd: str | list[str],
+    config: dict[str, Any],
+    abort_reasons: list[str] | None = None,
+) -> dict[str, Any]:
+    """
+    Execute a command with retries and optional abort conditions.
+
+    Args:
+        cmd (str | list[str]): Command to execute.
+        config (dict[str, Any]): Configuration dictionary for command execution.
+        abort_reasons (list[str] | None): List of reasons to abort execution if found in the output.
+
+    Returns:
+        dict[str, Any]: Status and output of the command execution.
+    """
     start_time = time.time()
 
     shell = config.get("shell", False)
     tries_total = config.get("retry_count", 0) + 1
     status = {"current_try": 0, "tries_total": tries_total, "output": []}
-    if isinstance(cmd, list):
-        proc_cmd = cmd[0]
-    else:
-        proc_cmd = os.path.basename(cmd.split(" ", maxsplit=1)[0])
-    for i in range(0, tries_total):
+    proc_cmd = (
```

Comment:
i don't really care too much, but this here is actually taking up more lines than the simple if/else.

i totally understand the urge for such a "rewrite". but when looking at it, it neither really improved readability nor line numbers..

but like i said, leave it. just an fyi